src/components/PluginHook.js

```diff
@@ -6,11 +6,15 @@ export const PluginHook = React.forwardRef((props, ref) => {
   const { classes, ...otherProps } = props; // eslint-disable-line react/prop-types
   return PluginComponents ? (
     PluginComponents.map((PluginComponent, index) => ( // eslint-disable-line react/prop-types
-      <PluginComponent
-        ref={ref}
-        {...otherProps}
-        key={index} // eslint-disable-line react/no-array-index-key
-      />
+      React.isValidElement(PluginComponent)
+        ? React.cloneElement(PluginComponent, { ...otherProps, ref })
```

Comment:
Do we need the distinguishing prop `key` here? Or will we expect that passed in? Or will it just be ok?

Reply:
I think not (and we probably don't need it below either) because it's totally artificial 🤷‍♂️